rasa_sdk/types.py

```diff
@@ -0,0 +1,58 @@
+from typing import Any, Dict, List, Optional, Text, Union
+
+from typing_extensions import TypedDict
+
+
+class TrackerState(TypedDict):
+    """
+    A dictionary representation of the state of a conversation.
+    """
+
+    # id of the source of the messages
+    sender_id: Text
+    # the currently set values of the slots
+    slots: Dict[Text, Any]
+    # the most recent message sent by the user
+    latest_message: Optional[Dict[Text, Any]]
+    # list of previously seen events
+    events: List[Dict[Text, Any]]
+    # whether the tracker is currently paused
+    paused: bool
+    # a deterministically scheduled action to be executed next
+    followup_action: Optional[Text]
+    # the loop that is currently active
+    active_loop: Dict[Text, Any]
+    # `active_form` is deprecated in favor of `active_loop`
+    active_form: Dict[Text, Any]
+    # the name of the previously executed action or text of e2e action
+    latest_action_name: Optional[Text]
+
+
+class DomainDict(TypedDict):
+    """
+    A dictionary representation of the domain.
+    """
+
+    intents: List[Union[Text, Dict[Text, Any]]]
```

Comment:
can this really contain strings?

Reply:
no you're right. The domain is implemented to handle both, but the dictionary representation of the domain only contains list of dicts in the `intents` key (cf. [here](https://github.com/RasaHQ/rasa/blob/05be168f7d3a68c64453b3c444cf4dff1eea1ce4/rasa/shared/core/domain.py#L825-L853))

Reply:
and it's the same for forms actually